Add error message for invalid loginless links.

eltigar opened the pull request:
Added proper error handling for invalid loginless authentication links.

Changes made:
- Added error handling in loginless function with Django messages framework
- Added conditional message display block in base template

What exactly triggers an error message:
- validate_token() returns None instead of a user object
- validation logic unchanged, including identical output for either invalid or expired token

eltigar committed:
Add error message for invalid loginless links.

marqueewinq commented on pairings/frontend/templates/frontend/base.html:
```diff
@@ -25,6 +25,18 @@
+    <div class="container-lg mt-3">
+        {% if messages %}
+            {% for message in messages %}
+            <div class="alert alert-danger border border-danger shadow-sm alert-dismissible fade show" role="alert">
```
This only allows for the error messages -- change the template to properly map the message type to the bootstrap style.

messages.error -> danger
messages.warning -> warning

etc.

eltigar committed:
Add mapping from the message type to the bootstrap style. Minor code optimizations.

eltigar commented:
I've added the mapping and tested other types to work locally. Also, I removed a redundant "if messages" condition and tested that it properly handles the case with no messages.

marqueewinq commented on pairings/accounts/viewsets.py:
```diff
@@ -55,4 +56,6 @@ def loginless(self, request, token):
         user = validate_token(token)
         if user is not None:
             login(request, user, backend=settings.AUTHENTICATION_BACKENDS[0])
+        else:
+            messages.error(request, "Login link is invalid or has expired. Please request a new one.")
         return redirect("index")
```
You might wanna cover this with e2e test, mocking message and redirect

marqueewinq reviewed (approved):
Add the test (or don't)
